consul/consul.go

```diff
@@ -44,6 +45,13 @@ func (c *Adapter) New(cfg *config.Config, podNodeName string, podIP string) *Ada
 		glog.Fatalf("bad adapter uri: ")
 	}
 
+	cfg.Consul.HttpClient = &http.Client{
```

Comment:
Why is it needed? Can't it be moved to the `default` in `switch` few lines below?

Reply:
Due to a change introduced in hashicorp/consul@c6d0b4cf907bbbe10404a55aa17bff339142fc8d, the HttpClient in the config returned by DefaultConfig() is nil. I will move it within the switch block.